juju/utils.py

```diff
@@ -139,7 +145,9 @@ async def _block():
 
 
 async def block_until_with_coroutine(
-    condition_coroutine, timeout=None, wait_period=0.5
+    condition_coroutine: Callable[[], Coroutine[Any, Any, bool]],
```

Comment:
What do you think about a type hint in this style instead?

```suggestion
    condition_coroutine: Callable[[], Awaitable[bool]],
```